### PR TITLE
Update snippets

### PR DESCRIPTION
**Pull request by SingularisArt**
Hello all.

This is one of my first pull requests. So, tell me if I'm doing anything wrong because I won't know.
Anywho, I updated the latex, lua, and gitcommit snippets.

For latex, I added a trillion more snippets and updated the current snippets.
For lua, I just added two more snippets. One for require and one for use.
For gitcommit, I removed all snippets and created 3 snippets.
    Instead of having different snippets like docs, refactor, test, etc I created a snippet that is based on python and while you type, it will give you that result.
    I added a signature snippet that will give you your signature.
    I added a good commit structure with details on what to say and what not to say as well as information on how to word and structure your sentences.

### CODE REVIEW

**Comment by lpil** [comment]
Hello! Would you mind putting back the delete snippets? Someone may be using them.

**Comment by SingularisArt** [comment]
Alright. I added the snippets that I removed from the gitcommit.snippets file.
Quick question. How do I add that commit to this pull request?

**Review by lpil — APPROVED** [review]
Thank you

**Comment by lygamac** [comment]
@SingularisArt can you remove the auto trigger tag for `use` snippet in `lua`? It gets triggered when typing common words like `user`

**Comment by SingularisArt** [comment]
Ah. Yes. I'll do that.

File: UltiSnips/tex.snippets [hunk 1]
```diff
@@ -1,258 +1,871 @@
...
+\chapter{$1${VISUAL}}
 endsnippet
 
-snippet lp "Long parenthesis"
-\left(${1:${VISUAL:contents}}\right)$0
+snippet sec "Section" wi
+\section{$1${VISUAL}}
 endsnippet
 
-snippet "mint(ed)?( (\S+))?" "Minted code typeset" br
```
Comment (sanjayankur31) [inline UltiSnips/tex.snippets:209]: Hi folks: was this an intentional removal?

**Comment by SingularisArt** [comment]
@sanjayankur31 Sorry for the headache I've caused. I'll keep that in mind when doing another PR.

**Comment by sanjayankur31** [comment]
No worries.

So was dropping the minted snippet intentional or was that just a fall out from the refactor? Were there other snippets that were dropped here, and were they all intentional?

I personally use minted quite a bit so I'd like to have the snippet, but I can keep it in a personal repo if folks think it's too specific to be here. If it should be here, I'll open a PR to add it back.

**Comment by ThomasFeher** [comment]
Why was the labeling for the section commands removed?

**Pull request merged**